### PR TITLE
Add a workflow-level error banner on the UI for errors that occur outside the workflow's context

### PR DESCRIPTION
**Pull request by kenxu95**
## Describe your changes and why you are making these changes

Also fixes a bug in `get_artifact_result` where the server could panic in certain failure conditions.

This is what it looks like. Test that it works against historical workflows too.
image

## Checklist before requesting a review
- [x] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [x] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [x] If this is a new feature, I have added unit tests and integration tests.
- [x] I have run the integration tests locally and they are passing.
- [x] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/ui/common/src/components/pages/workflow/id/index.tsx [hunk 1]
```diff
@@ -482,6 +482,25 @@ const WorkflowPage: React.FC<WorkflowPageProps> = ({
           </Box>
         )}
 
+        {/*Show any workflow-level errors at the top of the workflow details page.*/}
+        {workflow.selectedResult.exec_state.error !== null && (
```
Comment (vsreekanti) [inline src/ui/common/src/components/pages/workflow/id/index.tsx:486]: Tiny nit: You can just do `workflow.selectedResult.exec_state.error && (` — Typescript treats `null` and `undefined` as `False`.

File: src/ui/common/src/components/pages/workflow/id/index.tsx [hunk 2]
```diff
@@ -482,6 +482,25 @@ const WorkflowPage: React.FC<WorkflowPageProps> = ({
...
+          <Box
+            sx={{
+              backgroundColor: theme.palette.red[100],
+              color: theme.palette.red[600],
+              p: 2,
+              paddingBottom: '16px',
+              paddingTop: '16px',
+              height: 'fit-content',
+              width: !drawerIsOpen ? `calc(100% - 148px)` : 'calc(100% - 32px)',
```
Comment (vsreekanti) [inline src/ui/common/src/components/pages/workflow/id/index.tsx:495]: What are these numbers coming from? Presumably they should be `const`s somewhere if they're not already defined?

File: src/golang/cmd/server/handler/get_workflow.go [hunk 3]
```diff
@@ -187,10 +191,28 @@ func (h *GetWorkflowHandler) Perform(ctx context.Context, interfaceArgs interfac
 
 	workflowDagResults := make([]workflowDagResult, 0, len(dagResults))
 	for _, dagResult := range dagResults {
+		var dagExecState shared.ExecutionState
+		if !dagResult.ExecState.IsNull {
```
Comment (vsreekanti) [inline src/golang/cmd/server/handler/get_workflow.go:195]: What is the scenario in which the DAG result would have a different status vs. execution state? That seems very strange.

Comment (vsreekanti) [reply src/golang/cmd/server/handler/get_workflow.go:195]: It also doesn't seem like we're doing anything other than logging that information — is that intentional?

**Review by vsreekanti — APPROVED** [review]
Mostly looks good. A few stylistically things inline and a question about error handling on the backend.

File: src/golang/cmd/server/handler/get_workflow.go (hunk 3 above)
Comment (kenxu95) [reply src/golang/cmd/server/handler/get_workflow.go:195]: yeah it's mostly just a passive check. It definitely shouldn't happen but we don't technically validate this anywhere. Once https://linear.app/aqueducthq/issue/ENG-2665/remove-redundant-status-field-in-workflowdagresult-responses is addressed we won't need to think about this at all.

File: src/ui/common/src/components/pages/workflow/id/index.tsx (hunk 2 above)
Comment (kenxu95) [reply src/ui/common/src/components/pages/workflow/id/index.tsx:495]: I think they're hardcoded in this file. Can comment with a more thorough explanation.

File: src/golang/cmd/server/handler/get_workflow.go (hunk 3 above)
Comment (vsreekanti) [reply src/golang/cmd/server/handler/get_workflow.go:195]: Got it. Should we tag this to be removed once ENG-2665 is addressed?

Comment (kenxu95) [reply src/golang/cmd/server/handler/get_workflow.go:195]: yes, can do

**Comment by kenxu95** [comment]
Skipping tests since only a comment added.

**Pull request merged**